File: backend/app/Models/Almacen.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Almacen extends Model
{
    protected $table = 'almacenes';

    protected $fillable = [
        'nombre',
        'codigo',
        'direccion',
        'tipo',
        'activo',
    ];

    protected function casts(): array
    {
        return [
            'activo' => 'boolean',
        ];
    }

    public function stocks()
    {
        return $this->hasMany(ProductoAlmacenStock::class);
    }

    public function movimientos()
    {
        return $this->hasMany(MovimientoInventario::class);
    }

    public function transferenciasOrigen()
    {
        return $this->hasMany(Transferencia::class, 'almacen_origen_id');
    }

    public function transferenciasDestino()
    {
        return $this->hasMany(Transferencia::class, 'almacen_destino_id');
    }

    public function ajustes()
    {
        return $this->hasMany(AjusteInventario::class);
    }

    public function tomas()
    {
        return $this->hasMany(TomaInventario::class);
    }
}

File: app_abarrotes/lib/models/almacen.dart
class Almacen {
  final int id;
  final String nombre;
  final String codigo;
  final String tipo;
  final String? direccion;
  final bool activo;

  Almacen({
    required this.id,
    required this.nombre,
    required this.codigo,
    this.tipo = 'principal',
    this.direccion,
    required this.activo,
  });

  factory Almacen.fromJson(Map<String, dynamic> json) {
    return Almacen(
      id: json['id'],
      nombre: json['nombre'],
      codigo: json['codigo'],
      tipo: json['tipo'] ?? 'principal',
      direccion: json['direccion'],
      activo: json['activo'] ?? true,
    );
  }

  Map<String, dynamic> toJson() => {
    'nombre': nombre,
    'codigo': codigo,
    'tipo': tipo,
    'direccion': direccion,
    'activo': activo,
  };
}
